src/types.rs

```diff
@@ -24,16 +24,29 @@ const fn hex_val(c: u8) -> Option<u8> {
 
 macro_rules! key {
     ($i:ident, $s:expr) => {
-        #[derive(Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        #[repr(transparent)]
```

Comment:
why adding `transparent` here is required?

Reply:
I don't think its a good idea to mix the output index with the bytes representation. If its a matter of encoding / decoding, we can surely optimize this there but then not in the type representation and memory layout

Reply:
It is not a matter of encoding/decoding. `UtxoId` is an identifier as `MessageId` and `ContractId`, so it should have all methods available for other identifiers.

If we use representation:

```rust
pub struct UtxoId([u8; 32], u8);
```

We need to implement all traits manually. But `UtxoId(Bytes32, u8)` equevalent to `UtxoId([u8; 33])`. `AsRef<[u8]>` implementation for `UtxoId(Bytes32, u8)` requires usage of `unsafe { mem::transmute }` when it is natively available for `UtxoId([u8; 33])`.

Also, information about the output index matters during the creation of the `UtxoId`, but after it is used not often.

Reply:
I checked our implementation of "unsafe" methods, and it is required for memory layout correctness in our unsafe code